tools/truncate-silence.sh

```diff
@@ -0,0 +1,82 @@
+#!/bin/bash
+tools_path=$(dirname "$0")
+
+source "$tools_path/check-ffmpeg.sh"
+if ! check_ffmpeg; then
+    exit 1
+fi
+
+usage() {
+    echo "Usage: $0 [OPTION] [ARGS]" 1>&2;
+    echo "Truncates silence from the start or the end of an audio file"
+    echo "    --check-codecs    Checks if the encoders used by the HackSoundServer are installed." 1>&2
+    echo "    -i[:path]         Path of the input file." 1>&2
+    echo "    -s                Truncate silence from the start of the stream." 1>&2
+    echo "    -e                Truncate silence from the end of the stream." 1>&2
+    echo "    -r[:ratio]        Amplitude ratio. Samples with lower amplitude than this ratio will be removed." 1>&2
+    exit 1;
+}
+
+input_file=
+truncate_start=false
+truncate_end=false
+ratio=0.005
+show_help=true
+
+opts=$(getopt \
+    --longoptions "check-codecs" \
+    --name "$(basename "$0")" \
+    --options "cser:i:" \
+    -- "$@"
+)
+
+while true; do
+  case "$1" in
+    -c | --check-codecs)
+        show_help=false
+        if ! check_codecs; then
+            exit 1
+        fi
+        ;;
+    -s)
+        show_help=false
+        truncate_start=true
+        ;;
+    -e)
+        show_help=false
+        truncate_end=true
+        ;;
+    -r)
+        show_help=false
+        ratio="$2"
+        shift
+        ;;
+    -i)
+        show_help=false
+        input_file="$2"
+        shift
+        ;;
+    *)
+        break;;
+    esac
+    shift
+done;
+
+if ${show_help}; then
+    usage
+    exit 1
+fi
+if [ -z $input_file ]; then
```

Comment:
I'm not 100% sure about this, but I think it needs to be `"$input_file"` so that it evaluates to `""` and not nothing when the string is empty.